pkgs/applications/science/misc/gephi/default.nix

```diff
@@ -1,37 +1,64 @@
-{stdenv, fetchurl, jdk}:
+{ stdenv, fetchFromGitHub, callPackage, jdk, maven, javaPackages }:
 
-with stdenv.lib;
+let
+  version = "0.9.2";
 
-let version = "0.9.2"; in
-stdenv.mkDerivation {
-  name = "gephi-${version}";
-
-  src = fetchurl {
-    url = "https://github.com/gephi/gephi/releases/download/v${version}/gephi-${version}-linux.tar.gz";
-    sha256 = "1wr3rka8j2y10nnwbg27iaxkbrp4d7d07ccs9n94yqv6wqawj5z8";
+  src = fetchFromGitHub {
+    owner = "gephi";
+    repo = "gephi";
+    rev = "v${version}";
+    sha256 = "0kqp2nvnsb55j1axb6hk0mlw5alyaiyb70z0mdybhpqqxyw2da2r";
   };
 
-  meta = {
-    inherit version;
-    description = "A platform for visualizing and manipulating large graphs";
-    homepage = https://gephi.org;
-    license = licenses.gpl3;
-    maintainers = [maintainers.taeer];
-    platforms = platforms.linux;
+  # perform fake build to make a fixed-output derivation out of the files downloaded from maven central (120MB)
+  deps = stdenv.mkDerivation {
+    name = "gephi-${version}-deps";
+    inherit src;
+    buildInputs = [ jdk maven ];
+    buildPhase = ''
+      while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
```

Comment:
Should there be a an upper limit to not restart builds forever?

Reply:
It is copy/paste from hadoop derivation.
perhaps it is not needed here.
just in case, for people with home/mobile Internet.

Reply:
On Hydra there is a total timeout for a build, isn't there?
At home I'd prefer the forever loop until all files are downloaded, otherwise the case "maven have been restarted 20 times and has not downloaded all the files" would require some handling.
A forever wrapped around ```nix-build``` maybe.

Reply:
If put a limit there, what could be the number?
It is not retry number for each download (as it would be for curl), it is a loop around hundreds of downloads. 2-3 retries are "normal" even on Hydra on AWS (for ```hadoop``` derivation). Even with 10-20 could be false failures. 50 is virtually forever.

Reply:
If there is not internet connection at all or there is a problem with package build, it should fail comparable fast I suppose. So a high number should be still ok.

Reply:
Maybe "N iterations without showing progress" (with the same ```df $out```) would go